rust/src/io/exec/scan.rs

```diff
@@ -51,7 +51,7 @@ impl Scan {
     ) -> Self {
         let (tx, rx) = mpsc::channel(prefetch_size);
 
-        let projection = projection.clone();
+        let projection = Arc::new(projection.clone());
```

Comment:
is projection already `Arc<Schema>`? if so, dont need `Arc::new(..)` .

Reply:
Maybe try to run `cargo clippy` to see what reports are there in these changed lines?